airbyte-webapp/src/pages/SettingsPage/pages/NotificationPage/components/WebHookForm.tsx

```diff
@@ -1,95 +1,60 @@
 import { FormattedMessage, useIntl } from "react-intl";
-import styled from "styled-components";
 import * as yup from "yup";
 
 import { Label, LabeledSwitch } from "components";
 import { Row, Cell } from "components/SimpleTableComponents";
 import { Button } from "components/ui/Button";
 import { Input } from "components/ui/Input";
+import { Text } from "components/ui/Text";
+import { Tooltip } from "components/ui/Tooltip";
 
-import { WebhookPayload } from "hooks/services/useWorkspace";
-import { equal } from "utils/objects";
+import useWorkspace, { WebhookPayload } from "hooks/services/useWorkspace";
 
-const Text = styled.div`
-  font-style: normal;
-  font-weight: normal;
-  font-size: 13px;
-  line-height: 150%;
-  padding-bottom: 5px;
-`;
+import PlayIcon from "../../../../../components/icons/PlayIcon";
+import DocsIcon from "../../../../../views/layout/SideBar/components/DocsIcon";
+import { Content, SettingsCard } from "../../SettingsComponents";
+import styles from "./WebHookForm.module.scss";
 
-const InputRow = styled(Row)`
-  height: auto;
-  margin-bottom: 40px;
-`;
-
-const Message = styled(Text)`
-  margin: -40px 0 21px;
-  padding: 0;
-  color: ${({ theme }) => theme.greyColor40};
-`;
-
-const FeedbackCell = styled(Cell)`
-  &:last-child {
-    text-align: left;
-  }
-  padding-left: 11px;
-`;
+const enum WebhookAction {
+  Test = "test",
+  Save = "save",
+}
 
-const Success = styled.div`
-  font-size: 13px;
-  color: ${({ theme }) => theme.successColor};
-`;
+interface FormActionType {
+  [WebhookAction.Test]: boolean;
+  [WebhookAction.Save]: boolean;
+}
 
-const Error = styled(Success)`
-  color: ${({ theme }) => theme.dangerColor};
-`;
+interface WebHookFormProps {
+  webhook: WebhookPayload;
+}
 
 const webhookValidationSchema = yup.object().shape({
   webhook: yup.string().url("form.url.error"),
   sendOnSuccess: yup.boolean(),
   sendOnFailure: yup.boolean(),
 });
 
-interface WebHookFormProps {
-  webhook: WebhookPayload;
-  successMessage?: React.ReactNode;
-  errorMessage?: React.ReactNode;
-  onSubmit: (data: WebhookPayload) => void;
-  onTest: (data: WebhookPayload) => void;
-}
-
-const WebHookForm: React.FC<WebHookFormProps> = ({ webhook, onSubmit, successMessage, errorMessage, onTest }) => {
+export const WebHookForm: React.FC<WebHookFormProps> = ({ webhook }) => {
+  const [webhookViewGuide, setWebhookViewGuide] = useState(false);
+  const [formAction, setFormAction] = useState<FormActionType>({ test: false, save: false });
+  const { updateWebhook, testWebhook } = useWorkspace();
   const { formatMessage } = useIntl();
 
-  const feedBackBlock = (dirty: boolean, isSubmitting: boolean, webhook?: string) => {
-    if (successMessage) {
-      return <Success>{successMessage}</Success>;
+  const webhookChange = async (action: WebhookAction, data: WebhookPayload) => {
+    setFormAction((value) => ({ ...value, [action]: true }));
+    if (action === WebhookAction.Test) {
+      await testWebhook(data);
```

Comment:
There should be a feedback message when the notification test succeeds or fails, the same for trying to save the connection. There is no feedback message, even on the main branch. Would it be possible to fix this here?

Reply:
Agree with @edmundito. Also tested locally, and we indeed need to show the user the webhook URL testing (was it successful or not)

@matter-q yeah, we have such a service - https://github.com/airbytehq/airbyte/blob/9db329eb6c9e2cd3e7815b46fd2c651dfa3c3b14/airbyte-webapp/src/hooks/services/Notification/NotificationService.tsx#L12

Please note, there can be 2 possible cases when a testing hook request can be failed:
- 4xx status response from BE (if we have sent the totally wrong URL)
- 2xx status response from 3rd party hook service (need to check the response)

I've used the https://webhook.site/ service for testing webhooks